llvm/lib/Transforms/Vectorize/VPlanAnalysis.cpp

```diff
@@ -259,20 +259,17 @@ Type *VPTypeAnalysis::inferScalarType(const VPValue *V) {
                 VPPartialReductionRecipe>([this](const VPRecipeBase *R) {
             return inferScalarType(R->getOperand(0));
           })
+          .Case<VPInstructionWithType, VPWidenIntrinsicRecipe>(
+              [](const auto *R) { return R->getResultType(); })
```

Comment:
Worth noting that this case should appear before the next to catch VPInstructionWithType before its parent VPInstruction?